okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -33,7 +33,10 @@
 final class RealCall implements Call {
   final OkHttpClient client;
   final RetryAndFollowUpInterceptor retryAndFollowUpInterceptor;
-  final EventListener eventListener;
+
+  // There is a cycle between the Call and EventListener that makes this awkward. This will be set
+  // after we create the call instance then create the event listener instance.
+  volatile EventListener eventListener;
```

Comment:
> This code is single-threaded, so volatile and safe-publication aren’t relevant?

Ahh, yeah my bad. I'm conflating 2 separate issues here.

**Issue 1**: The current API can (theoretically) be used improperly. 
Sounds like we're ok with just documenting this fact. Worst case is to enforce it.

**Issue 2**: Should EventListener have volatile.
Per your note above, I removed `volatile`.